src/module.c

```diff
@@ -10388,6 +10382,148 @@ int RM_GetDbIdFromDefragCtx(RedisModuleDefragCtx *ctx) {
     return ctx->dbid;
 }
 
+/* --------------------------------------------------------------------------
+ * ## Modules EventLoop API
+ * --------------------------------------------------------------------------*/
+
+typedef struct EventLoopData {
+    RedisModuleEventLoopCallback rProc;
+    RedisModuleEventLoopCallback wProc;
+    void *user_data;
+} EventLoopData;
+
+static int eventLoopToAeMask(int mask) {
+    int aeMask = 0;
+    if (mask & REDISMODULE_EVENTLOOP_READABLE)
+        aeMask |= AE_READABLE;
+    if (mask & REDISMODULE_EVENTLOOP_WRITABLE)
+        aeMask |= AE_WRITABLE;
+    return aeMask;
+}
+
+static int eventLoopFromAeMask(int ae_mask) {
+    int mask = 0;
+    if (ae_mask & AE_READABLE)
+        mask |= REDISMODULE_EVENTLOOP_READABLE;
+    if (ae_mask & AE_WRITABLE)
+        mask |= REDISMODULE_EVENTLOOP_WRITABLE;
+    return mask;
+}
+
+static void eventLoopCbReadable(struct aeEventLoop *ae, int fd, void *user_data, int ae_mask) {
+    UNUSED(ae);
+    EventLoopData *data = user_data;
+    data->rProc(fd, data->user_data, eventLoopFromAeMask(ae_mask));
+}
+
+static void eventLoopCbWritable(struct aeEventLoop *ae, int fd, void *user_data, int ae_mask) {
+    UNUSED(ae);
+    EventLoopData *data = user_data;
+    data->wProc(fd, data->user_data, eventLoopFromAeMask(ae_mask));
+}
+
+/* Add an event to the event loop
+ *
+ * On success REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned and errno is set to the following values:
+ *
+ * * ERANGE: 'fd' is negative or higher than 'maxclients' Redis config.
+ * * EINVAL: 'callback' is NULL or 'mask' value is invalid.
+ *            Valid values for 'mask':
+ *               REDISMODULE_EVENTLOOP_READABLE
+ *               REDISMODULE_EVENTLOOP_WRITABLE
+ *               REDISMODULE_EVENTLOOP_READABLE | REDISMODULE_EVENTLOOP_WRITABLE
```

Comment:
is this a good markdown? (i'm not sure)

Reply:
If you meant flags, removed them. Please take a look

Reply:
i don't see you changed anything.
what i was concerned about is whatever this will be correctly rendered by the markdown in redis.io.
@zuiderkwast you're the expert for that.

Reply:
It will probably not be rendered very well. A blank line creates a new paragraph but a single linebreak will just be discarded, so I think this will all be displayed on a single line and look like this:

* ERANGE: 'fd' is negative or higher than 'maxclients' Redis config.
* EINVAL: 'callback' is NULL or 'mask' value is invalid. Valid values for 'mask': REDISMODULE_EVENTLOOP_READABLE REDISMODULE_EVENTLOOP_WRITABLE REDISMODULE_EVENTLOOP_READABLE | REDISMODULE_EVENTLOOP_WRITABLE

I don't really know how you want to display it. A code block or a nested list?

If you want a code block, you need to indent 4 spaces more than the bullet list and add a black line before and after.

To be really sure, it's possible to preview the docs using `ruby src/modules/gendoc.rb`, directing the output to the overwrite the corresponding markdown file in the redis-doc repo and then running the redis-io website locally to preview how it looks.... (Too much work?)

Reply:
Oh sorry, I completely misunderstood first. I didn't know about redis.io 
I'm taking a look, will deploy locally and check it. Thank you both.